types/AST.js

```diff
@@ -260,8 +260,10 @@ type Func = {
   // Only in WAST
   name: ?Index,
 
-  params: Array<FuncParam>,
-  result: Array<Valtype>,
+  params?: Array<FuncParam>,
+  result?: Array<Valtype>,
+  typeRef?: Identifier | NumberLiteral,
```

Comment:
Is there a better type to use for this? If not, perhaps we should define a type for this ...

```
type TypeRef = Identifier | NumberLiteral
```

Reply:
Yes, there's `Index` https://github.com/xtuc/webassemblyjs/blob/1e06bbd2b3a6117ceca12efc7ed99ef652e4b741/types/AST.js#L46-L54